numpy/random/mtrand/mtrand.pyx

```diff
@@ -1426,15 +1435,272 @@ cdef class RandomState:
         >>> plt.show()
 
         """
+        if sizeof(long) == sizeof(npy_int64):
+            return self.random_int64(low, high, size)
+        elif sizeof(long) == sizeof(npy_int32):
+            return self.random_int32(low, high, size)
+        else:
+            msg = 'Unimplemented for sizeof(long) == %d' % sizeof(long)
+            raise RuntimeError(msg)
+
+    def random_intp(self, low, high=None, size=None):
```

Comment:
That is true.  I wonder then if you could put logic in ```random_integers``` to check the OS and Python version and then call the appropriate one (either the ```np.intp``` or ```long``` versions) to maintain backwards compatibility?  Then the user wouldn't have to worry / think about which one to call.  This would require abstracting the current code in ```random_integers``` into say a function called ```rand_int``` (placeholder, this name may not work so well as it would get confusing with ```randint```; though perhaps the names might need to be refactored).